src/config.rs

```diff
@@ -3,7 +3,7 @@
 use std::time::Duration;
 
 /// Indicates whether only the provided directory or its sub-directories as well should be watched
-#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug)]
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
 pub enum RecursiveMode {
```

Comment:
For this and Config I can't think of a reason to stuff them (indirectly) into a hashmap, but ok.